File: Keepr/Models/Vault.cs
namespace Keepr.Models
{
    public class Vault : RepoItem<int>
    {
        public Profile Creator { get; set; }
        public string CreatorId { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public string Img { get; set;}
        public bool IsPrivate { get; set; }
    }
}

File: Keepr.client/src/models/Vault.js
import { Profile } from "./Account";

export class Vault {
    constructor(data) {
        this.id = data.id
        this.name = data.name
        this.description = data.description
        this.img = data.img
        this.isPrivate = data.isPrivate
        this.creatorId = data.creatorId
        this.creator = data.creator ? new Profile(data.creator) : null
        this.createdAt = new Date(data.createdAt)
        this.updatedAt = new Date(data.updatedAt)
    }
}